bagitobjecttransfer/recordtransfer/views.py

```diff
@@ -536,32 +558,14 @@ def get_submission_group(self, cleaned_form_data: dict) -> Optional[SubmissionGr
         """
         group = None
 
-        group_name = cleaned_form_data["group_name"]
-        if group_name != "No Group":
-            if group_name == "Add New Group":
-                new_group_name = cleaned_form_data["new_group_name"]
-                description = cleaned_form_data["group_description"]
-
-                group, created = SubmissionGroup.objects.get_or_create(
-                    name=new_group_name,
-                    description=description,
-                    created_by=self.request.user,
-                )
-                if created:
-                    LOGGER.info('Created "%s" SubmissionGroup', new_group_name)
-
+        group_id = cleaned_form_data["group_id"]
+        if group_id:
+            try:
+                group = SubmissionGroup.objects.get(uuid=group_id, created_by=self.request.user)
                 LOGGER.info('Associating Submission with "%s" SubmissionGroup', group.name)
 
-            else:
-                try:
-                    group = SubmissionGroup.objects.get(
-                        name=group_name, created_by=self.request.user
-                    )
-                    LOGGER.info('Associating Submission with "%s" SubmissionGroup', group.name)
-
-                except SubmissionGroup.DoesNotExist as exc:
-                    LOGGER.error('Could not find "%s" SubmissionGroup', group.name, exc_info=exc)
-                    group = None
+            except SubmissionGroup.DoesNotExist as exc:
+                LOGGER.error('Could not find SubmissionGroup with ID %d', group_id, exc_info=exc)
```

Comment:
Small tweak to error message (since it's a UUID, not an ID, which someone might confuse for a primary key).

Also, this will cause an error because `%d` is for numerical data, and `group_id` is a string now.

```suggestion
                LOGGER.error('Could not find SubmissionGroup with UUID %s', group_id, exc_info=exc)
```

Reply:
Good catch.

Resolved by:
https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/226/commits/80ff0ff2cb064cd3099c525980ff554d3c2a2627